pelicun/model/damage_model.py

```diff
@@ -1308,38 +1384,12 @@ def map_ds(values: np.ndarray, offset: int) -> np.ndarray:
         if not scaling_specification:
             scaling_specification = {}
         else:
-            # if there are contents, ensure they are valid.
-            # See docstring for an example of what is expected.
-            parsed_scaling_specification = {}
-            # validate contents
-            for key, value in scaling_specification.items():
-                css = 'capacity adjustment specification'
-                if not isinstance(value, str):
-                    msg = (
-                        f'Invalid entry in {css}: {value}. It has to be a string. '
-                        f'See docstring of DamageModel._create_dmg_RVs.'
-                    )
-                    raise TypeError(msg)
-                capacity_adjustment_operation = value[0]
-                number = value[1::]
-                if capacity_adjustment_operation not in {'+', '-', '*', '/'}:
-                    msg = (
-                        f'Invalid operation in {css}: '
-                        f'{capacity_adjustment_operation}'
-                    )
-                    raise ValueError(msg)
-                fnumber = base.float_or_None(number)
-                if fnumber is None:
-                    msg = f'Invalid number in {css}: {number}'
-                    raise ValueError(msg)
-                parsed_scaling_specification[key] = (
-                    capacity_adjustment_operation,
-                    fnumber,
-                )
-                scaling_specification = parsed_scaling_specification
+            scaling_specification = parse_scaling_specification(
+                scaling_specification
+            )
 
         # get the component sample and blocks from the asset model
-        for pg in pgb.index:
+        for pg in pgb.index:  # noqa: PLR1702
```

Comment:
I wonder how this PLR1702 came up for you and not for John and I. When I look it up it says "This rule is unstable and in [preview](https://docs.astral.sh/ruff/preview/). The --preview flag is required for use.", see here: https://docs.astral.sh/ruff/rules/too-many-nested-blocks/

@ioannis-vm , I would appreciate your advice on this one.

Reply:
I ran ruff again without --preview. The error is still there. I am not sure what a better option is than breaking this error using noqa. So, I am leaving this noqa here. Please advise if you guys have better ideas.